orderer/consensus/hlmirbft/chain.go

```diff
@@ -591,6 +592,84 @@ func (c *Chain) writeBlock(block *common.Block, index uint64) {
 	c.support.WriteBlock(block, m)
 }
 
+
+//JIRA FLY2-103 :function to get the config metadata from envelope payload
+func(c *Chain) getConfigMetadata(msgPayload []byte) (*hlmirbft.ConfigMetadata, error) {
+	payload, err := protoutil.UnmarshalPayload(msgPayload)
+	if err != nil {
+		return nil,err
+	}
+	// get config update
+	configUpdate, err := configtx.UnmarshalConfigUpdateFromPayload(payload)
+	if err != nil {
+		return nil,err
+	}
+	return  MetadataFromConfigUpdate(configUpdate)
+
+}
+
+//JIRA FLY2-103 : function to generate new network state config
+func (c *Chain) getNewNetworkStateConfig(newNodeList []uint64) *msgs.NetworkState_Config  {
+	nodes := newNodeList
+	nodeCount := len(nodes)
+	numberOfBuckets := int32(nodeCount)
+	checkpointInterval := numberOfBuckets * 5
+	maxEpochLength := uint64(checkpointInterval * 10)
+	return &msgs.NetworkState_Config{
+			Nodes:              nodes,
+			MaxEpochLength:     maxEpochLength,
+			CheckpointInterval: checkpointInterval,
+			F:                  int32((nodeCount - 1) / 3),
+			NumberOfBuckets:    numberOfBuckets,
+		}
+}
+
+//JIRA FLY2-103 : function to identify the type of config update and return the config change
+func (c *Chain) getUpdatedConfigChange(configMetaData *hlmirbft.ConfigMetadata,currentConsenters , updatedConsenters []*hlmirbft.Consenter) ([]*msgs.Reconfiguration,error){
```

Comment:
Could you add options to the configMetadata which can be used in the new network state config? e.g MaxEpochLength, CheckpointInterval, NumberOfBuckets. Afterwards, make sure to update the fabric-protos-go of this branch.